File: frontend/src/lib/pdnConsent.ts
/**
 * The personal-data consent text this build displays, by its issue date.
 *
 * DEFINED TWICE, ON PURPOSE, AND CHECKED NOWHERE — like the protocol frames and the backup
 * format. The other half is ServerDotnet/Shared/Outcome.Abstractions/Legal/PdnConsent.cs, and
 * the text itself is frontend/public/pdn.html. All three move in one commit or none.
 *
 * The client sends this with the registration, and the server records what it was sent rather
 * than stamping its own: an older build showing an older text should be recorded honestly as
 * that older text, not silently upgraded to whatever is published today.
 */
export const PDN_CONSENT_VERSION = "2026-08-21";

/** Where the text lives. Static HTML, so it opens even when the app itself is broken. */
export const PDN_CONSENT_URL = "/pdn.html#soglasie";

/** The policy — what the operator does with the data. Published under art. 18.1, and NOT part
 *  of the consent: since 156-ФЗ (1 Sep 2025) consent has to stand apart from other documents,
 *  so it is linked separately rather than folded into the same sentence. */
export const PDN_POLICY_URL = "/pdn.html";

/**
 * Whether OUR consent text is the right thing to show against `host`.
 *
 * The text names a specific operator — us. On a server somebody else runs, the operator is
 * them: asking there would not be an extra safeguard, it would put our name on their data
 * processing and collect a consent that means nothing. This client can sign into any instance,
 * so that is the common case rather than an edge one.
 *
 * Tenant spaces on *.outcome.ru are our infrastructure and count. An empty host means the
 * instance that served this page, which on our own deployment is us.
 *
 * Mirrors pdnConsentApplies() in mobile/lib/pdn_consent.dart — the two must agree, or the same
 * account sees the box on one client and not the other.
 */
export function pdnConsentApplies(host: string): boolean {
  const h = host.trim().toLowerCase()
    .replace(/^https?:\/\//, "")
    .split("/")[0]!
    .split(":")[0]!;
  if (h === "") return typeof window !== "undefined"
    && (window.location.hostname === "outcome.ru"
        || window.location.hostname.endsWith(".outcome.ru"));
  return h === "outcome.ru" || h.endsWith(".outcome.ru");
}

File: ServerDotnet/Shared/Outcome.Abstractions/Legal/PdnConsent.cs
namespace Outcome.Shared.Legal;

/// <summary>
/// The personal-data consent text currently published, identified by the date it was issued.
///
/// Russian law (152-ФЗ) makes an operator of anyone processing personal data — an email address
/// alone is enough — and the operator has to be able to show that consent was obtained. A
/// checkbox that leaves no trace shows nothing, so every account records the moment and the
/// version; see <c>User.ConsentAt</c> / <c>ConsentVersion</c>.
///
/// WHEN THE TEXT CHANGES, change this string in the same commit as the document. They are a
/// pair that has to move together: a version naming a text nobody can produce is worse than no
/// version at all, and a reworded text under an unchanged version silently rewrites what past
/// users are recorded as having agreed to.
/// </summary>
public static class PdnConsent
{
    /// <summary>Issue date of the published text, ISO-8601. Recorded against every new account.</summary>
    public const string Version = "2026-08-21";

    /// <summary>Where the text lives, for clients that need to link to it.</summary>
    public const string Path = "/pdn.html";
}
